src/shared/components/Banners/Banners.tsx

```diff
@@ -1,21 +1,29 @@
 import { Collapse } from '@mui/material';
 import { TransitionGroup } from 'react-transition-group';
 
-import { BannerComponents, banners } from 'shared/state/Banners';
+import { banners } from 'shared/state/Banners';
 import { useAppDispatch } from 'redux/store';
 
+import { BannerComponents } from './Banners.const';
+
 export const Banners = () => {
   const dispatch = useAppDispatch();
   const data = banners.useData();
 
   return (
     <TransitionGroup>
-      {data.banners.map(({ key }) => {
+      {data.banners.map(({ key, bannerProps }) => {
         const BannerComponent = BannerComponents[key];
 
         return (
           <Collapse key={key}>
-            <BannerComponent onClose={() => dispatch(banners.actions.removeBanner({ key }))} />
+            <BannerComponent
+              {...bannerProps}
+              onClose={() => {
```

Comment:
As a common rule, we use an anonymous function into the html as a callback handler for cases when it includes one callback/action. In this case I would move the logic into a separate function. It reduces html code and improves readability. But feel free discuss it of course.